plat/kvm/arm/entry64.S

```diff
@@ -57,15 +59,22 @@ ENTRY(_libkvmplat_entry)
 	isb
 #endif /* CONFIG_FPSIMD */
 
-	/*
-	 * We will disable MMU and cache before the pagetables are ready.
-	 * This means we will change memory with cache disabled, so we need to
-	 * invalidate the cache to ensure there is no stale data in it.
-	 * But we don't know the size of the RAM either. And it would be
-	 * expensive to invalidate the whole cache. In this case, just
-	 * just need to invalidate what we are going to use:
-	 * DTB, TEXT, DATA, BSS, and bootstack.
+#ifndef CONFIG_OPTIMIZE_PIE
```

Comment:
Does `OPTIMIZE_PIE` imply that the Linux Boot Protocol is used? At a first glance this seems unrelated.

Reply:
It's unrelated indeed (also notice this is an `ifndef`). I do not make an explicit check for `CONFIG_KVM_BOOT_PROTO_LXBOOT` but rely on the MMU heuristic to also cover for QEMU virt, and potentially other platforms that don't use the linux boot protocol but exhibit the same behavior. Tbh I'm between two minds about this. On the one hand implicit detection looks like a bad idea, on the other hand it allows more flexibility. If we want to be on the safe side I can make this conditional to `BOOT_PROTO_LXBOOT` OR `BOOT_PROTO_QEMU_VIRT`.
What do you think?